source/Core/LangSupport/lang_multi.cpp

```diff
@@ -0,0 +1,24 @@
+#include "../../configuration.h"
+#include "OLED.hpp"
+#include "Translation.h"
+#include "gui.hpp"
+
+const char *translatedString(const char *const *strs) {
+  if (systemSettings.uiLanguage >= LanguageCount) {
+    systemSettings.uiLanguage = 0;
+  }
+  return strs[systemSettings.uiLanguage];
+}
+
+bool settings_setLanguageSwitch(void) {
```

Comment:
If we are going to start splitting apart the settings, maybe its worth splitting the other ones into files based on their menu?

Reply:
Hmm, I split them because when I used a conditional to check `LanguageCount`, the LTO process didn't seem to be smart enough to cleanly eliminate the check and inline the call on single-language firmwares. (But the effect isn't really that big.)

I wasn't aiming to restructure the gui.cpp file.